File: c/1187.c
#include <stdio.h>
#include <string.h>
int main(){
    double m[12][12], soma=0;
    char s;
    int i, j;

    scanf("%c", &s);
    for (i=0;i<12;i++){
        for (j=0;j<12;j++){
            scanf("%lf", &m[i][j]);
        }
    }

    for (i=0;i<12;i++){
        for (j=0;j<12;j++){
            if (j+i <=10 && j>i && j!=i){
                soma+=m[i][j];
            }
        }
    }

    if (s=='S'){
        printf("%.1lf\n", soma);
    }else{
        printf("%.1lf\n", soma/30);
    }
}

File: python/1187.py
s = str(input())
matriz = []
for i in range(0,12):
    linha = []
    for j in range(0,12):
        linha.append(float(input()))
    matriz.append(linha)
soma = 0

for i in range(0,12):
    for j in range(0,12):
        if j+i <=10 and j>i and j!=i:
            soma+=matriz[i][j]
if s=='S':
    print(f"{soma:.1f}")
elif s=='M':
    soma/=30
    print(f"{soma:.1f}")